### ISSUE TITLE
Crash when recording avi with a lua script using GlideN64 (and gln)

### ISSUE THREAD

**Issue by Wade7wastaken** [mkdasher/mupen64-rr-lua-#168]
**Describe the bug**
Crash occurs when resuming the emulator after starting avi recording while a lua script is running (tested with [this](https://github.com/Mupen64-Rewrite/SM64LuaRedux/releases/tag/1.7.0) script). This crash only occurs when using GlideN64, glN64, (maybe others). Works with Jabo's.

Visual Studio shows the crash in vcr.cpp
![image](https://github.com/mkdasher/mupen64-rr-lua-/assets/86390605/e882fab1-fed4-4c57-96ab-256735735969)

**To Reproduce**
1. Use GlideN64 or glN64
2. Run a lua script (pretty sure it has to do some kind of drawing but im not sure)
3. Start avi recording
4. Unpause the emulator
5. Crash

**Expected behavior**
To not crash and record both the gameplay and lua script

**Comment by Aurumaker72** [comment]
It's crashing inside free()

GLideN64 doesn't provide a free override (DllCrtFree), so mupen falls back to the standard library one, assuming the plugin is using it as well. 
At this point, mupen crashes. Either GLideN64 isn't using it to allocate the image buffer after all, or there's a race condition-induced double free somewhere. Will keep you posted

**Comment by Aurumaker72** [comment]
GLideN64 implementation:

```cpp
void DisplayWindowWindows::_readScreen(void **_pDest, long *_pWidth, long *_pHeight)
{
	*_pWidth = m_width;
	*_pHeight = m_height;

	*_pDest = malloc(m_height * m_width * 3);
...
```

It's using /MT, just like mupen, so CRT should be compatible.

**Comment by Aurumaker72** [comment]
Adding a DllCrtFree implementation to GLideN64 moves the issue elsewhere, now crashing inside mupen's vfw call with an AV, specifically in `vcr_compress.c:193`

The video stream is initialized with an incorrect size of 864x480, but the image buffer acquired from `readScreen` is (rightly) 640x480.

**Comment by Aurumaker72** [comment]
DllCrtFree can be found automatically, and @CasualPokePlayer wrote an implementation for this.

This issue's root is fixed, but the other crash remains.  It only happens when a window-resizing lua script is running. 
The vcr compressor attempts to read from the readscreen buffer and assumes it has the same size as the window's game area, which fails catastrophically as it has been resized.

**Comment by Aurumaker72** [comment]
Closed by 340727d
